integration_tests/src/main/python/cache_test.py

```diff
@@ -285,3 +285,15 @@ def helper(spark):
         return df.selectExpr("a")
 
     assert_gpu_and_cpu_are_equal_collect(helper)
+
+
+@pytest.mark.skipif(is_before_spark_330(), reason='DayTimeInterval is not supported before Spark3.3.0')
+@pytest.mark.parametrize('enable_vectorized_conf', enable_vectorized_confs, ids=idfn)
+@ignore_order(local=True)
+def test_cache_daytimeinterval(enable_vectorized_conf):
```

Comment:
Now, it begs a question, do we need a separate test or can we add an additional type to one (all?) of the existing test? 

You can use a pytest.param(..., marks=[skipif]) and add it to an existing test. The case may be that you need an additional test, I am not sure

Reply:
I tried this `pytest.param(..., marks=[skipif])` but got the below error on Spark before 3.3.0.

```

==================================== ERRORS ====================================
________________ ERROR collecting src/main/python/cache_test.py ________________
../../src/main/python/cache_test.py:60: in <module>
    daytime_interval_gen = pytest.param(DayTimeIntervalGen(),
../../src/main/python/data_gen.py:626: in __init__
    super().__init__(DayTimeIntervalType(), nullable=nullable, special_cases=special_cases)
E   NameError: name 'DayTimeIntervalType' is not defined
- generated xml file: /home/liangcail/work/projects/on_github/spark-rapids/integration_tests/target/run_dir/TEST-pytest-1647331437827840861.xml -
=========================== short test summary info ============================
ERROR ../../src/main/python/cache_test.py - NameError: name 'DayTimeIntervalT...
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
====================== 11715 deselected, 1 error in 2.51s ======================

```